crates/astria-core/src/sequencer/v1alpha1/block.rs

```diff
@@ -565,3 +568,77 @@ where
     };
     merkle::Tree::from_leaves(iter.into_iter().map(|item| Sha256::digest(&item)))
 }
+
+/// [`Deposit`] represents a deposit from the sequencer to a rollup.
+///
+/// A [`Deposit`] is constructed whenever a [`BridgeLockAction`] is executed
+/// and stored as part of the block's events.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct Deposit {
+    pub bridge_address: Address,
+    pub rollup_id: RollupId,
+    pub amount: u128,
+    pub asset_id: asset::Id,
+    pub destination_chain_address: String,
+}
+
+impl Deposit {
+    #[must_use]
+    pub fn into_raw(self) -> raw::Deposit {
+        let Self {
+            bridge_address,
+            rollup_id,
+            amount,
+            asset_id,
+            destination_chain_address,
+        } = self;
+        raw::Deposit {
+            bridge_address: bridge_address.to_vec(),
+            rollup_id: rollup_id.to_vec(),
+            amount: Some(amount.into()),
+            asset_id: asset_id.as_bytes().to_vec(),
+            destination_chain_address,
+        }
+    }
+
+    /// Attempts to transform the deposit from its raw representation.
+    ///
+    /// # Errors
+    ///
+    /// - if the bridge address is invalid
+    /// - if the amount is unset
+    /// - if the rollup ID is invalid
+    /// - if the asset ID is invalid
+    pub fn try_from_raw(raw: raw::Deposit) -> Result<Self, DepositError> {
+        let raw::Deposit {
+            bridge_address,
+            rollup_id,
+            amount,
+            asset_id,
+            destination_chain_address,
+        } = raw;
+        let bridge_address = Address::try_from_slice(&bridge_address)?;
+        let amount = amount.ok_or(DepositError::FieldNotSet("amount"))?.into();
+        let rollup_id = RollupId::try_from_slice(&rollup_id)?;
+        let asset_id = asset::Id::try_from_slice(&asset_id)?;
+        Ok(Self {
+            bridge_address,
+            rollup_id,
+            amount,
+            asset_id,
+            destination_chain_address,
+        })
+    }
+}
+
+#[derive(Debug, thiserror::Error)]
+pub enum DepositError {
```

Comment:
This library tries to not make the inner errors part of its public interface. Please make this a newtype wrapper: `pub struct DepositError(DepositErrorKind)` and then define a private `enum DepositErrorKind`